src/coreclr/src/gc/gc.cpp

```diff
@@ -5839,6 +5844,17 @@ bool gc_heap::virtual_commit (void* address, size_t size, gc_oh_num oh, int h_nu
     assert (heap_hard_limit == 0);
 #endif //!HOST_64BIT
 
+#ifdef VERIFY_COMMITTED_BY_OH
+#ifdef MULTIPLE_HEAPS
+    if (h_number != -1)
```

Comment:
>if (h_number != -1) [](start = 4, length = 19)

is it intentional that this check is only done in Server case? because Workstation does the same thing - it also passes -1 for the same situations that Server does